lib/appExtensions.ts

```diff
@@ -0,0 +1,221 @@
+  storeHash: string,
+  accessToken: string,
+  label: {
+    defaultValue: string;
+    locales: Array<{ value: string; localeCode: string }>;
+  }
+) {
+  const response = await fetch(
+    `https://api.bigcommerce.com/stores/${storeHash}/graphql`,
+
+    {
+      method: "POST",
+      headers: {
+        accept: "application/json",
+        "content-type": "application/json",
+        "x-auth-token": accessToken,
+      },
+      body: JSON.stringify({
+        query: `
+            mutation AppExtension($input: CreateAppExtensionInput!) {
+                appExtension {
+                  createAppExtension(input: $input) {
+                    appExtension {
+                      id
+                      context
+                      label {
+                        defaultValue
+                        locales {
+                          value
+                          localeCode
+                        }
+                      }
+                      model
+                      url
+                    }
+                  }
+                }
+              }`,
+        variables: {
+          input: {
+            context: "PANEL",
+            model: "PRODUCTS",
+            url: "/productAppExtension/${id}",
+            label: label,
+          },
+        },
+      }),
+    }
+  );
+
+  return response;
+}
+
+// Checks for current App Extensions
+export async function checkAppExtensionsExist(
+  storeHash: string,
+  accessToken: string,
+  expectedAppExtensions: Array<{
+    defaultValue: string;
+    locales: Array<{ value: string; localeCode: string }>;
+  }>
+) {
+  // Get current App Extensions
+  const currentAppExtensions = await getAppExtensions(storeHash, accessToken);
+
+  // Get Labels of Current App Extensions
+  const currentAppExtensionLabels = await currentAppExtensions.map(
+    (edge: any) => edge.node.label
+  );
+
+  for (const label of expectedAppExtensions) {
+    if (!currentAppExtensionLabels.includes(label)) {
+      await createAppExtensionOnLoad(storeHash, accessToken, label);
+    }
+  }
+}
+
+// Gets all current App Extensions installed on store
+export async function getAppExtensions(storeHash: string, accessToken: string) {
+  const response = await fetch(
+    `https://api.bigcommerce.com/stores/${storeHash}/graphql`,
+    {
+      method: "POST",
+      headers: {
+        accept: "application/json",
+        "content-type": "application/json",
+        "x-auth-token": accessToken,
+      },
+      body: JSON.stringify({
+        query: `
+                    query {
+                        store {
+                            appExtensions {
+                                edges {
+                                    node {
+                                        id
+                                    }
+                                }
+                            }
+                        }
+                    }`,
+      }),
+    }
+  );
+  if (!response.ok) {
```

Comment:
🍹 Is there a reason we're handling errors differently for `get` than for `delete`? I think the way `delete` is handled would be preferred here because it includes the error message from the server, which would help for debugging.

Reply:
Updated to match error handling in get